### PR TITLE
chore: update config for running new guest auth tests

### PR DESCRIPTION
**Pull request by reesscot**
#### Description of changes

This PR adds the CircleCI config to run integration tests for the guest authentication user scenario.

#### Description of how you validated changes
Tested against CircleCI, and tests ran successfully here: 
https://app.circleci.com/pipelines/github/aws-amplify/amplify-js/7967/workflows/85b88d47-595c-4ba5-99e9-3af77b619811

```
[1]   Guest to Auth user scenario:
[1]     ✓ Unauthenticated (guest) Identity ID should be initially created in localstorage, then cleared after sign in (4457ms)
[1]     ✓ Authenticated user tokens should be created in localstorage on signin, removed on signout (2543ms)
[1] 
[1] 
[1]   2 passing (7s)
[1] 
[1] 
[1]   (Results)
[1] 
[1]   ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
[1]   │ Tests:        2                                                                                │
[1]   │ Passing:      2                                                                                │
[1]   │ Failing:      0                                                                                │
[1]   │ Pending:      0                                                                                │
[1]   │ Skipped:      0                                                                                │
[1]   │ Screenshots:  0                                                                                │
[1]   │ Video:        true                                                                             │
[1]   │ Duration:     7 seconds                                                                        │
[1]   │ Spec Ran:     auth/guest-to-auth-user.spec.ts                                                  │
[1]   └────────────────────────────────────────────────────────────────────────────────────────────────┘

```

#### Checklist

- [x] PR description included

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -383,6 +383,13 @@ jobs:
           sample_name: amplify-authenticator
           spec: ui-amplify-authenticator
           browser: << parameters.browser >>
+      - integ_test_js:
+          test_name: 'React Authenticator'
```
Comment (reesscot) [inline .circleci/config.yml:387]: Wasn't sure how the categories should work. Would it be better to name this "Guest to Authenticated User"?

Comment (amhinson) [reply .circleci/config.yml:387]: Yea, thats a good thought here to add a bit more clarity 👍

Comment (reesscot) [reply .circleci/config.yml]: Cool, updated

**Review by slaymance — APPROVED** [review]
Looks good! Verified that the tests ran and passed in CircleCI.

**Review by amhinson — APPROVED** [review]
LGTM 👍

**Pull request merged**